### PR TITLE
gmt5, gmt6: deprecate gmt5, fix sighandler and cmake

### PR DESCRIPTION
**Pull request by remkos**
#### Description

* `gmt5` and `gmt6` Portfiles have been separated as a better forward approach to eventually removing `gmt5` (https://trac.macports.org/ticket/71306 and PR #26601) as the latter is marked obsolete since the maintainers are already for 5 years on `gmt6`.
* `gmt5` had complications getting compiled on MacOS since the latest revbump (https://trac.macports.org/ticket/71306): this has been fixed with a new patch.
* `gmt5` and `gmt6` Portfiles have been updated to the latest standards.

This PR thus makes PR #26601 obsolete.
See https://trac.macports.org/ticket/71306

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 15.1 24B83 x86_64
Command Line Tools 16.1.0.0.1.1729049160

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by remkos** [comment]
Please note that the gmt6 package passed on all platforms, but not the gmt5 ones.
This is a known issue since #26601 (see also https://trac.macports.org/ticket/71306).

I suggest this PR is accepted so that at least `gmt6` works, picking up the sighandler update.
Then some can continue working on `gmt5`, although we now have marked it obsolete.

File: science/gmt6/Portfile [hunk 1]
```diff
@@ -0,0 +1,147 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           cmake 1.0
```
Comment (reneeotten) [inline science/gmt6/Portfile:4]: this should probably use the latest version of the PortGroup (which is 1.1)

File: science/gmt6/Portfile [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                    port:ghostscript \
+                    port:gshhg-gmt \
+                    port:netcdf
+
+default_variants    +gdal +pcre
+if {![variant_isset lgpl]} {
+    default_variants-append +nonfree
+}
+
+cmake.out_of_source yes
```
Comment (reneeotten) [inline science/gmt6/Portfile:55]: this can be removed as it is the default in tue latest PortGroup version.

File: science/gmt6/Portfile [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+# Mimic CMake's default FLAGS:
+if {[variant_isset debug]} {
+    configure.optflags      -O0
+} else {
+    # optflags deliberately unset:
+    configure.optflags
+    # Set CMAKE_BUILD_TYPE=RelWithDebInfo to get reliable backtraces:
+    configure.args-delete   -DCMAKE_BUILD_TYPE=Release
+    configure.args-append   -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (reneeotten) [inline science/gmt6/Portfile:66]: this could/should be ....-replace

File: science/gmt6/Portfile [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+    foreach l {gmt postscriptlight} {
+        file delete ${destroot}${prefix}/lib/${subport}/bin/lib${l}.${version}.dylib
+    }
+    ln -s ${prefix}/lib/${subport}/bin/gmt ${destroot}${prefix}/bin/${subport}
+}
+
+variant fftw3 description {Optional support for FFTW-3 library} {
+    depends_lib-append      port:fftw-3-single
+    configure.args-delete   -DFFTW3_ROOT=off
+    configure.args-append   -DFFTW3_ROOT=${prefix}
```
Comment (reneeotten) [inline science/gmt6/Portfile:95]: also ...-replace, please check remainder of the Portfile for this as well.

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: science/gmt6/Portfile (hunk 2 above)
Comment (remkos) [reply science/gmt6/Portfile:55]: Removed in commit  c70ef3c

Comment (reneeotten) [reply science/gmt6/Portfile:55]: It's still there

Comment (remkos) [reply science/gmt6/Portfile]: Oops. You're right. I fixed it in gmt5/Portfile but overlooked it in gmt6/Portfile.
Fixed now in [3b87f0f](https://github.com/macports/macports-ports/commit/3b87f0fd3cc6db67fe5f3fbf67f9ce0a59c21dc3)

**Comment by reneeotten** [comment]
thanks @remkos!

**Pull request merged**